### PR TITLE
Format markdown & add better display for examples

### PR DESCRIPTION
**Pull request by juraj-hrivnak**
- Formated markdown.
- Replaced all headers in examples with `!!! example`.
- Added better example for Minecraft recipe builders.
- Increased font size in example blocks to be the same as regular code blocks.
- Added option to autohide the main header when you scroll down.

### CODE REVIEW

File: mkdocs.yml [hunk 1]
```diff
@@ -260,6 +260,7 @@ theme:
         name: Switch to dark mode
 
   features:
+    - header.autohide
```
Comment (WaitingIdly) [inline mkdocs.yml]: when the header autohides, the navigation bar doesnt move up with it, leaving a rather large area weirdly empty. is this possible to fix?

File: docs/groovyscript/mods/thaumcraft/Infusion.md [hunk 2]
```diff
@@ -55,7 +55,7 @@ Register recipe: (returns nothing)
 .register()
 ```
 
-### Example
+!!! example
 
 ```groovy
```
Comment (WaitingIdly) [inline docs/groovyscript/mods/thaumcraft/Infusion.md]: needs to be indented, doesnt display properly currently

**Review by WaitingIdly — COMMENTED** [review]
a few of examples werent converted due to not being a header or being non-standard text. They include a few things in the vanilla minecraft folder, including `ore_dictionary.md`, `tooltip.md`, and `aa_events.md`, along with most things in the gregtech folder.

File: mkdocs.yml (hunk 1 above)
Comment (juraj-hrivnak) [reply mkdocs.yml]: I've removed this option for now. I will make a separate PR when I figure it out.

**Comment by juraj-hrivnak** [comment]
Converted the remaining examples in `ore_dictionary.md`, `tooltip.md`, `aa_events.md` and those I missed.
And added a copy-to-clipboard button for code blocks.

**Review by brachy84 — APPROVED** [review]

**Pull request merged**